### DISCUSSION TITLE
Connector does not response with expected messages

### DISCUSSION THREAD

**Discussion by reisman234** [eclipse-edc/Samples#106] [Q&A]
I'm using the transfer-06-consumer-pull-http example as a basis for my own connector with extensions which I require for my use case. At the moment, I have just added the provision-http extensions.

```
implementation("org.eclipse.edc:provision-http:0.1.0")
```

When the Connector sends the provision request to my Provisioner, It makes some request to the connector back, for example with these requests against the connector the Provisioner wants to know the ID of the consumer. To get this information, it firsts requests for the`transferprosesses` which the provisioner is currently processing. This DataObject holds the `contractId` information, which then is used to request for the ContractAgreement DataObject, which has the desired information about the Consumer.

Now, when the provisioner does the first request to get the Transferprocess DataObject, I can see in the logs that the following data was returned. 

```
provisioner - DEBUG - url: http://localhost:19193/management/v2/transferprocesses/d284e01e-f83b-4e6f-bc42-99d82f68613a 
{
  "@id": "d284e01e-f83b-4e6f-bc42-99d82f68613a",
  "@type": "edc:TransferProcess",
  "edc:correlationId": "742fb10c-f103-47ba-8d0c-dc204c003b4a",
  "edc:state": "PROVISIONING_REQUESTED",
  "edc:stateTimestamp": 1693817773129,
  "edc:type": "PROVIDER",
  "edc:assetId": "assetId",
  "edc:contractId": "MQ==:YXNzZXRJZA==:MTM1NDQwZWItY2NkNC00MGMyLThhNzItMmFiODdjYmEwNjAx",
  "edc:callbackAddresses": [],
  "edc:dataDestination": {
    "@type": "edc:DataAddress",
    "edc:type": "HttpProxy"
  },
  "@context": {
    "dct": "https://purl.org/dc/terms/",
    "edc": "https://w3id.org/edc/v0.0.1/ns/",
    "dcat": "https://www.w3.org/ns/dcat/",
    "odrl": "http://www.w3.org/ns/odrl/2/",
    "dspace": "https://w3id.org/dspace/v0.8/"
  }
}
```
The received data is parsed into a TransferRequestDto Model, which I have generated from the openapi scheme. But now errors occur because the received data does not match the model.

1. Why is in each key name a `edc:` prefix? This is not in the api schema and for my understanding a mismatch between schema and data.
    - none of the schemas for the different versions shows this structure with a leading `edc:` in its key name. 
3. What is `@type": "edc:TransferProcess"`, is this really the type of the received data scheme? 
    - this type is introduced in version 0.1.4-SNAPSHOT but build with version 0.1.1 (`gradle.properties`)
4. With which version of the EDC dependencies is the connector build? From my understanding of grade, it should be the version `0.1.0`, which is defined in the `gradle.properties`.

**Comment by reisman234** [comment]
> 3\. With which version of the EDC dependencies is the connector build? From my understanding of grade, it should be the version `0.1.0`, which is defined in the `gradle.properties`

I have figured out that I get with `./gradlew transfer:transfer-06-consumer-pull-http:http-pull-connector:dependencies` a complete list of the used dependencies with its versions. 

Version 0.2.1 is used

**Comment by ndr-brt** [comment]
> 1. Why is in each key name a `edc:` prefix? This is not in the api schema and for my understanding a mismatch between schema and data.
>
>    * none of the schemas for the different versions shows this structure with a leading `edc:` in its key name.

EDC follows Json-LD semantics, I suggest to take a look at the [Json-LD specs](https://www.w3.org/TR/json-ld11/) for details. In every case, further suggestion is to always "expand" the json documents returned by EDC before evaluating them.
The openapi schema reports a particular "compacted" view of the json-ld representation, that does not necessary match with the actual returned by EDC. By expanding them you'll get the match. 

> 2. What is `@type": "edc:TransferProcess"`, is this really the type of the received data scheme?
>    
>    * this type is introduced in version 0.1.4-SNAPSHOT but build with version 0.1.1 (`gradle.properties`)

Every type that the EDC produces has a `@type` definition. That one is a `TransferProcess`.

> 3. With which version of the EDC dependencies is the connector build? From my understanding of grade, it should be the version `0.1.0`, which is defined in the `gradle.properties`.

Samples should be aligned with the latest version of the EDC, currently at `0.2.1`

**Reply by reisman234** [reply]
> Samples should be aligned with the latest version of the EDC, currently at 0.2.1

I think, I don't understand gradle :disappointed: ... The `edcVersion` Attribute, which is specified in the `gradle.properties` does nothing have to do, which Version of EDC is used during build :confused:

**Reply by ndr-brt** [reply]
that one determines the version of the `edc-build` plugin used, not the version of the EDC.
That was a leftover that indeed needs to use the EDC version provided by the gradle version catalog (`gradle/libs.versions.toml`). Will provide an issue/PR to fix it
